### PR TITLE
Add `cook_rgidf()` to pretend non-RGI with RGI format

### PR DESCRIPTION
**Pull request by Keeptg**
I add a function `cook_rgidf()` in `utils._funcs` to make it easier of using a non-RGI glacier inventory in OGGM.
Will add a tutorial to show how it can be used in future!

- [x] Tests added/passed
- [x] Fully documented
- [x] Entry in `whats-new.rst`

### CODE REVIEW

**Comment by Keeptg** [comment]
Not sure what's the reason the test program failed. Any suggestions wil be appriciated!

**Comment by fmaussion** [comment]
The tests fail because the test requires to download input data which is not in the test data, a situation we try to avoid. I think we don't need to test that everything works in your tests, let me have a look at your code later today.

File: oggm/tests/test_utils.py [hunk 1]
```diff
@@ -323,6 +323,26 @@ def test_clips_performance(self):
...
+        cfg.initialize()
+        cfg.PARAMS['use_intersects'] = False
+        cfg.PATHS['working_dir'] = utils.mkdir(utils.get_temp_dir(dirname='test_cook_rgidf'),
+                                               reset=True)
+        path = utils.get_demo_file('cgi2.shp')
+        cgidf = gpd.read_file(path)
+        rgidf = utils.cook_rgidf(cgidf, save_special_columns={'Glc_Long': 'CenLon',
+                                                              'Glc_Lati': 'CenLat'},
+                                 assign_col_values={'Area': cgidf.Glc_Area*1e-6})
+        gdirs = workflow.init_glacier_directories(rgidf)
```
Comment (fmaussion) [inline oggm/tests/test_utils.py:337]: Yes so I think you need to stop the test here, and ensure that OGGM could still create the glacier directories properly (that's the main test anyway)

File: oggm/utils/_funcs.py [hunk 2]
```diff
@@ -780,3 +781,96 @@ def shape_factor_adhikari(widths, heights, is_rectangular):
...
+               assign_col_values=None):
+    """ Cook the user's glacier inventory with the RGI format
+
+    Parameters
+    ----------
+    gi_gdf : :py:geopandas.GeoDataFrame
+        the GeoDataFrame of the user's glacier inventory
+    region : str
+        Glacier RGI region code, only usable when set `confirm_region=False`.
+        The default is '13'.
```
Comment (fmaussion) [inline oggm/utils/_funcs.py:797]: There should be no default here - it should fail if the region is not provided and cannot be inferred automatically

File: oggm/utils/_funcs.py [hunk 3]
```diff
@@ -780,3 +781,96 @@ def shape_factor_adhikari(widths, heights, is_rectangular):
...
+    version : str
+        Glacier inventory version code. The default is '60'.
```
Comment (fmaussion) [inline oggm/utils/_funcs.py:799]: Explain that this is necessary to generate the RGI Ids

File: oggm/utils/_funcs.py [hunk 4]
```diff
@@ -780,3 +781,96 @@ def shape_factor_adhikari(widths, heights, is_rectangular):
...
+            'Area': -9999, 'Zmin': -9999, 'Zmax': -9999, 'Zmed': -9999, 'Slope': -9999,
+            'Aspect': -9999, 'Lmax': -9999, 'Status': 0, 'Connect': 0, 'Form': 0,
+            'TermType': 0, 'Surging': 0, 'Linkages': 1, 'check_geom': None, 'Name': ''}
+
+    # Construct the output GeoDataFrame
+    cooked_rgidf = gpd.GeoDataFrame(data=data, geometry=geom, crs='epsg:4326')
+
+    # If there are specifical column in the original glacier inventory we want to keep
+    if save_special_columns is not None:
+        for col in save_special_columns.keys():
```
Comment (fmaussion) [inline oggm/utils/_funcs.py:868]: loop like:
`for key, val in dict.items()`

File: oggm/utils/_funcs.py [hunk 5]
```diff
@@ -780,3 +781,96 @@ def shape_factor_adhikari(widths, heights, is_rectangular):
...
+        Glacier RGI region code, only usable when set `confirm_region=False`.
+        The default is '13'.
+    version : str
+        Glacier inventory version code. The default is '60'.
+    ids : list of integer, each element should be a str of length 5
+        Assign special id number to each glacier. The default is None, the id code
+            following the glacier order.
+    id_suffix : str or None
+         Add a suffix to the glacier id. The default is None, no suffix
+    save_special_columns : dict or None
```
Comment (fmaussion) [inline oggm/utils/_funcs.py:805]: could this be only one argument (`assign_col_values`) where the right column name in the RGI format is already used? The users can change the name themselves if they want to, I'm not sure this dict here is very useful.

**Review by fmaussion — CHANGES_REQUESTED** [review]
Thanks! This is looking good. Some minor suggestions to the names, otherwise this looks good!

File: oggm/utils/_funcs.py (hunk 5 above)
Comment (Keeptg) [reply oggm/utils/_funcs.py:806]: OK! `assign_col_values` argument was deleted!

File: oggm/utils/_funcs.py (hunk 4 above)
Comment (Keeptg) [reply oggm/utils/_funcs.py]: Thanks! Didn't know this before.

**Comment by Keeptg** [comment]
There are still two error reported.
Don't know how to deal with them......

**Review by fmaussion — APPROVED** [review]

**Comment by fmaussion** [comment]
Thanks @Keeptg ! sorry this took so long, excellent work!

**Pull request merged**